src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+            for (int i = 0; i < items.Length; i++)
+            {
+                list.Remove(items[i]);
+                list.Remove(items[i]);
+                CollectionAssert.Equal(items.Slice(i + 1), list);
+            }
+        }
+
+        [Fact]
+        public void RemoveItems()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            var items = GenerateItems(16);
+            var list = GetList(items);
+            for (int i = 0; i < items.Length; i++)
+            {
+                list.Remove(items[i]);
+                CollectionAssert.Equal(items.Slice(i + 1), list);
+            }
+        }
+
+        [Fact]
+        public void RemoveNull()
+        {
+            if (ExpectedIsFixedSize || ExpectedIsReadOnly)
+                return;
+            if (ItemsMustBeNonNull)
+                return;
+            var items = new object[15];
+            var list = GetList(null);
+            var tempItems = GenerateItems(16);
+            list.AddRange(tempItems);
+            Array.Copy(tempItems, 0, items, 0, 11);
+            Array.Copy(tempItems, 12, items, 11, 4);
+            list.Insert(8, null);
+            list.Remove(tempItems[11]);
+            list.Remove(null);
+            list.Remove(GenerateItem());
+            CollectionAssert.Equal(items, list);
+
+            items = new object[15];
+            list = GetList(null);
+            tempItems = GenerateItems(16);
+            Array.Copy(tempItems, 0, items, 0, 11);
+            Array.Copy(tempItems, 12, items, 11, 4);
+            list.AddRange(tempItems);
+            list.Remove(tempItems[11]);
+            list.Remove(null);
+            list.Remove(GenerateItem());
+            CollectionAssert.Equal(items, list);
+        }
+
+        [Fact]
+        public void RemoveInvalidValues()
+        {
+            if (!IsGenericCompatibility)
+                return;
+            Type[] expectedExceptions = ExpectedIsFixedSize
+                                            ? new[]
+                                            {
+                                                typeof (
+                                                  ArgumentException),
+                                                typeof (
+                                                  NotSupportedException)
+                                            }
+                                            : new[]
+                                            {
+                                                typeof (
+                                                  ArgumentException)
+                                            };
+
+            var items = GenerateItems(16);
+            var list = GetList(items);
+            foreach (var i in GetInvalidValues())
+            {
+                var invalid = i;
+                if (ExpectedIsFixedSize
+                    || SearchingThrowsFromInvalidValue)
+                {
```

Comment:
I don't think it causes much noise. And the test is explicit about what it is testing. Other tests may do this as part of testing other things.